packages/browser/src/Session.ts

```diff
@@ -57,22 +57,17 @@ async function silentlyAuthenticate(
   clientAuthn: ClientAuthentication
 ) {
   // Check if we have an ID Token in storage - if we do then we may be
-  // currently logged in, and the user has refreshed their browser page. In
-  // this case, it can be really useful to save the user's current browser
-  // location, so that we can restore that location after we complete the
-  // entire re-login-the-user-silently-flow (e.g., if the user was on a
-  // specific 'page' of a Single Page App, then presumably they'll expect to
-  // be brought back to exactly that same 'page' after the full refresh).
-  const storedSessionInfo = await clientAuthn.getSessionInfo(sessionId);
-  if (
-    storedSessionInfo === undefined ||
-    storedSessionInfo.idToken === undefined
-  ) {
-    return;
-  }
-  window.localStorage.setItem(KEY_CURRENT_URL, window.location.href);
+  // currently logged in, and the user has refreshed their browser page. The ID
+  // token is validated, and its issuer claim is extracted to know where silent
+  // authentication is possible.
   const issuer = await clientAuthn.getCurrentIssuer();
-  if (issuer !== null) {
+  const storedSessionInfo = await clientAuthn.getSessionInfo(sessionId);
+  if (issuer !== null && storedSessionInfo !== undefined) {
```

Comment:
What would it mean here if one of these conditions was true, but the other wasn't? Is that worth trying to explain in a comment here...? I'm also thinking in terms of error handling and debugging this in the field - i.e., if someone reports that silent authentication isn't working - could we easily tell why...?

Reply:
Actually, thinking to answer this question lead me to refactor this portion a bit: it was redundant, because both `validateCurrentIssuer` and `getSessionInfo` looked up the same session information in storage. 26ad6902e4ab31ed2e972d7f47ecc61bafa4ece4 makes it clearer.